Sources/CryptomatorCloudAccess/Dropbox/DropboxCloudProvider.swift

```diff
@@ -192,7 +194,11 @@ public class DropboxCloudProvider: CloudProvider {
 				}
 				if let networkError = networkError {
 					CloudAccessDDLogDebug("DropboxCloudProvider: fetchItemMetadata(at: \(cloudPath.path)) failed with networkError: \(networkError)")
-					reject(self.convertRequestErrorToDropboxError(networkError))
+					if networkError.isAuthError() {
+						reject(CloudProviderError.unauthorized)
+					} else {
+						reject(self.convertRequestErrorToDropboxError(networkError))
+					}
```

Comment:
This seems oddly specific here. This change only affects `fetchItemMetadata()`. But I think we have to do a refactoring here, if you'd like to map "auth error" to our own unauthorized error. My suggestion is the following:

Rename `convertRequestErrorToDropboxError()` to `convertRequestError()` and it should just return a simple `Error` and not `DropboxError` anymore. And then for `error.isAuthError()`, it should return `CloudProviderError.unauthorized`. You can then remove `authError` from `DropboxError` completely.

Reply:
Thank you for the suggestion and guidance. I’ve refactored everything in commit 0c1a348.